src/hypertable.c

```diff
@@ -2499,6 +2508,9 @@ ts_hypertable_osm_range_update(PG_FUNCTION_ARGS)
 	Oid time_type; /* required for resolving the argument types, should match the hypertable
 					  partitioning column type */
 
+	/* this function is not meant to be run on a read-only secondary */
+	Assert(!RecoveryInProgress());
+
```

Comment:
For your future self, could you extend the comment to describe _why_ we can't reach this point in production builds?